src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinResultString = '';
+
+    if (this.length < 0) {
+      return joinResultString;
+    }
+
+    let workSeparator = separator;
+
+    if (workSeparator === null) {
+      workSeparator = 'null';
+    }
```

Comment:
```suggestion
```

I really have no idea, why would you need this. Just use `separator` from arguments

Reply:
https://prnt.sc/1jt64yn
one of the tests doesn't work without this